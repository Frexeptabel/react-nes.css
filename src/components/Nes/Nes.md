Nes-Icons example:

```js
<Nes character="squirtle" />
<Nes character="logo" />
<Nes character="logo" snes/>
```
