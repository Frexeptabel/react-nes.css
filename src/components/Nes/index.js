import React from "react";
import PropTypes from "prop-types";

const Nes = ({ character, snes, ...props }) => {
  const className = `${snes ? "s" : ""}nes-${character}`;
  return <i className={className} {...props} />;
};

Nes.propTypes = {
  character: PropTypes.oneOf([
    "octocat",
    "mario",
    "ash",
    "pokeball",
    "bulbasaur",
    "squirtle",
    "smartphone",
    "phone",
    "kirby",
    "logo",
    "jp-logo"
  ]).isRequired,
  snes: PropTypes.bool
};
export default Nes;
