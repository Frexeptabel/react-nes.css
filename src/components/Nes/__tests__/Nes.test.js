import React from "react";
import Nes from "../index";
import renderer from "react-test-renderer";

test("The Nes should be rendered properly", () => {
  const component = renderer.create(<Nes character="logo" />);
  expect(component).toMatchSnapshot();
});

test("The Nes should be rendered properly with snes prop", () => {
  const component = renderer.create(<Nes character="logo" snes />);
  expect(component).toMatchSnapshot();
});
