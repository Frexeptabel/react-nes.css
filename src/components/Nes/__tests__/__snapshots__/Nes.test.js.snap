// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Nes should be rendered properly 1`] = `
<i
  className="undefinednes-logo"
/>
`;

exports[`The Nes should be rendered properly with snes prop 1`] = `
<i
  className="snes-logo"
/>
`;
